unittests/test-contracts/db_find_secondary_test/db_find_secondary_test.wast

```diff
@@ -0,0 +1,120 @@
+      (drop (call $db_idx_long_double_store (local.get $receiver) (local.get $receiver) (local.get $receiver) (i64.const 8)  (i32.const 48)))
+      (drop (call $db_idx_long_double_store (local.get $receiver) (local.get $receiver) (local.get $receiver) (i64.const 1)  (i32.const 48)))
+      (drop (call $db_idx_long_double_store (local.get $receiver) (local.get $receiver) (local.get $receiver) (i64.const 4)  (i32.const 48)))
+      (drop (call $db_idx_long_double_store (local.get $receiver) (local.get $receiver) (local.get $receiver) (i64.const 90) (i32.const 48)))
+      (drop (call $db_idx_long_double_store (local.get $receiver) (local.get $receiver) (local.get $receiver) (i64.const 42) (i32.const 48)))
+      (drop (call $db_idx_long_double_store (local.get $receiver) (local.get $receiver) (local.get $receiver) (i64.const 3)  (i32.const 48)))
+      (drop (call $db_idx_long_double_store (local.get $receiver) (local.get $receiver) (local.get $receiver) (i64.const 60) (i32.const 48)))
+      ;;load 2 items with sec key at 80
+      (drop (call $db_idx_long_double_store (local.get $receiver) (local.get $receiver) (local.get $receiver) (i64.const 0)  (i32.const 80)))
+      (drop (call $db_idx_long_double_store (local.get $receiver) (local.get $receiver) (local.get $receiver) (i64.const 88) (i32.const 80)))
+
+      ;;test each index type. expect to find primary key of 1 when searching seckey at 48 since 1 was the lowest that was stored for the
+      ;; searched seckey (the primary key will be placed at memory location 1024)
+      (drop (call $db_idx64_find_secondary (local.get $receiver) (local.get $receiver) (local.get $receiver) (i32.const 48) (i32.const 1024)))
+      (if (i64.ne (i64.load (i32.const 1024)) (i64.const 1)) (then unreachable))
+      (drop (call $db_idx128_find_secondary (local.get $receiver) (local.get $receiver) (local.get $receiver) (i32.const 48) (i32.const 1024)))
+      (if (i64.ne (i64.load (i32.const 1024)) (i64.const 1)) (then unreachable))
+      (drop (call $db_idx256_find_secondary (local.get $receiver) (local.get $receiver) (local.get $receiver) (i32.const 48) (i32.const 2) (i32.const 1024)))
+      (if (i64.ne (i64.load (i32.const 1024)) (i64.const 1)) (then unreachable))
+      (drop (call $db_idx_double_find_secondary (local.get $receiver) (local.get $receiver) (local.get $receiver) (i32.const 48) (i32.const 1024)))
+      (if (i64.ne (i64.load (i32.const 1024)) (i64.const 1)) (then unreachable))
+      (drop (call $db_idx_long_double_find_secondary (local.get $receiver) (local.get $receiver) (local.get $receiver) (i32.const 48) (i32.const 1024)))
+      (if (i64.ne (i64.load (i32.const 1024)) (i64.const 1)) (then unreachable))
+   )
+
+   ;;secondary keys to be used
+   (data (i32.const 16) "\00\00\00\00\01\02\03\04AbcdefghABCDEFGH01234567")
+   (data (i32.const 48) "\00\00\00\00\01\02\03\05abcdefghABCDEFGH99999999")
+   (data (i32.const 80) "\00\00\00\00\01\02\03\06aBcdefghABCDEFGH00000000")
+)
```

Comment:
hm maybe I should have just done this in a .cpp contract, but I would have had to use the `internal_use_do_not_use` functions which I wanted to avoid.